src/controllers/validate.js

```diff
@@ -0,0 +1,87 @@
+const validateName = (name) => {
+  if (!name) {
+    throw new TypeError('Name is required');
+  }
+  if (typeof name !== 'string') {
+    throw new TypeError('This is not a name');
+  }
+  if (/\W/.test(name)) {
+    throw new TypeError('Name must contain alphanumeric characters');
+  }
+  if (name.length <= 1) {
+    throw new TypeError('Please enter your fullname');
+  }
+  if (name.length > 20) {
+    throw new TypeError('You have exceeded our maximum amount of characters');
+  }
+};
+
+const validateEmail = (email) => {
+  if(!email) {
+    throw new TypeError('Email is required');
+  }
+  if (typeof email !== 'string') {
+    throw new TypeError('This is not a valid email');
+  }
+  if(!email.includes('@') {
+    throw new TypeError('This is not a valid email');
+  }
+}
+
+const validatePassword = (password) => {
+  if(!password) {
+    throw new TypeError('Please insert a password');
+  }
+  if (typeof password !== 'string') {
+    throw new TypeError('This is not a valid password');
+  }
+  if (password.length < 8) {
+    throw new TypeError('Your password is too short');
+  }
+  const passwordStrengthRegex = ('^(?=.[a-z])(?=.[A-Z])(?=.[0-9])(?=.[!@#\$%\^&*])');
+  if(!passwordStrengthRegex.test(password)){
+    throw new TypeError('Your password needs to contain at least one uppercase, one lowercase, one special character and one number');
```

Comment:
Not sure I like [password restrictions](https://arstechnica.com/information-technology/2013/06/password-complexity-rules-more-annoying-less-effective-than-length-ones/) like this, what does everyone else think?

Reply:
Thanks for the tips @dangerdak. The changes have been made 👍